Data/Scripts/GardenConquest/Messaging/ResponseProcessor.cs

```diff
@@ -164,6 +202,99 @@ public class ResponseProcessor {
 			}
 		}
 
+		private void processFleetResponse(FleetResponse resp) {
+			log("Loading fleet data from server", "processFleetResponse");
+			List<GridEnforcer.GridData> gridData = resp.FleetData;
+
+			// Clear our current data to get fresh data from server
+			Array.Clear(m_Counts, 0, m_Counts.Length);
+			m_SupportedGrids.Clear();
+			m_UnsupportedGrids.Clear();
+
+			// Saving data from server to client
+			for (int i = 0; i < gridData.Count; ++i) {
+				int classID = (int)gridData[i].shipClass;
+				m_Counts[classID] += 1;
+				if (gridData[i].supported) {
+					if (!m_SupportedGrids.ContainsKey(classID)) {
+						m_SupportedGrids[classID] = new List<GridEnforcer.GridData>();
+					}
+					m_SupportedGrids[classID].Add(gridData[i]);
+				}
+				else {
+					if (!m_UnsupportedGrids.ContainsKey(classID)) {
+						m_UnsupportedGrids[classID] = new List<GridEnforcer.GridData>();
+					}
+					m_UnsupportedGrids[classID].Add(gridData[i]);
+				}
+			}
+
+			// Building fleet info to display in a dialog
+			string fleetInfoBody = buildFleetInfoBody(resp.OwnerType);
+			string fleetInfoTitle = buildFleetInfoTitle(resp.OwnerType);
+
+			// Displaying the fleet information
+			Utility.showDialog(fleetInfoTitle, fleetInfoBody, "Close");
+		}
+
+		#endregion
+		#region Process Response Utilities
+
+		private string buildFleetInfoBody(GridOwner.OWNER_TYPE ownerType) {
+			log("Building Fleet Info Body", "buildFleetInfoBody");
+			string fleetInfoBody = "";
+			List<GridEnforcer.GridData> gdList;
+			for (int i = 0; i < m_Counts.Length; ++i) {
+				if (m_Counts[i] > 0) {
+					fleetInfoBody += (HullClass.CLASS)i + ": " + m_Counts[i] + " / ";
+					if (ownerType == GridOwner.OWNER_TYPE.FACTION) {
+						fleetInfoBody += ServerSettings.HullRules[i].MaxPerFaction + "\n";
+					}
+					else if (ownerType == GridOwner.OWNER_TYPE.PLAYER) {
+						fleetInfoBody += ServerSettings.HullRules[i].MaxPerSoloPlayer + "\n";
+					}
+					else {
+						fleetInfoBody += "0\n";
+					}
+				}
+				if (m_SupportedGrids.ContainsKey(i)) {
+					gdList = m_SupportedGrids[i];
+					for (int j = 0; j < gdList.Count; ++j) {
+						fleetInfoBody += "  " + j + ". " + gdList[j].shipName + " - " + gdList[j].blockCount + " blocks\n";
+						if (gdList[j].displayPos) {
+							fleetInfoBody += "      GPS: " + gdList[j].shipPosition.X + ", " + gdList[j].shipPosition.Y + ", " + gdList[j].shipPosition.Z + "\n";
+						}
```

Comment:
If we can't display the position for the grid, let's instead display info on that i.e. "Position Unavailable - must own the Main Cockpit."